src/test/testSerialization.cpp

```diff
@@ -262,10 +263,13 @@ BOOST_FIXTURE_TEST_CASE(BaseSaveLoad, RandWorldFixture)
             std::vector<PlayerInfo> players;
             for(unsigned j = 0; j < 4; j++)
                 players.push_back(PlayerInfo(loadSave.GetPlayer(j)));
-            GlobalGameSettings newGGS = save.ggs;
+            GlobalGameSettings& newGGS = save.ggs;
             TestEventManager newEm(loadSave.start_gf);
             GameWorld newWorld(players, newGGS, newEm);
-            save.sgd.ReadSnapshot(newWorld);
+            Game newGame(newGGS, loadSave.start_gf, players);
+            boost::shared_ptr<Game> sharedGame;
+            sharedGame.reset(&newGame);
```

Comment:
This is not how a shared ptr works. Most importantly you need an actual pointer, so `new Game(...)` or make the `shared_ptr` not delete it's pointer.
`make_shared` is simply a wrapper around `new`. So what you want is: `boost::shared_ptr<Game> sharedGame = boost::make_shared<Game>(newGGS, loadSave.start_gf, players)` or `boost::shared_ptr<Game> sharedGame(new Game(newGGS, loadSave.start_gf, players))`

Reply:
Yeah i was just playing around with compiler errors. im not an c/c++ expert, so boost magic confuses me more and i was happy that it compiles at this point.

Reply:
no worries. `shared_ptr` is part of C++ now, so it's worth learning. All those `smart_ptr`s are just wrappers around a raw pointer to automatically delete the pointer when no one references it anymore.
But you can just do what you think should work and amend your last commit and we go from there.